src/s2geography/geoarrow.h

```diff
@@ -69,6 +78,44 @@ class Reader {
   std::unique_ptr<ReaderImpl> impl_;
 };
 
+class ExportOptions : public TessellationOptions {
+ public:
+  ExportOptions()
+      : TessellationOptions(),
+        significant_digits_(6) {}
+  int significant_digits() const { return significant_digits_; }
+  void set_significant_digits(int significant_digits) { significant_digits_ = significant_digits; }
+
+ private:
+  int significant_digits_;
+};
+
+class WriterImpl;
+
+/// \brief Array writer for any GeoArrow extension array
+///
+/// This class is used to convert Geography objects into an ArrowArray
+/// with geoarrow data (serialized or native).
+class Writer {
+ public:
+  enum class OutputType { kWKT, kWKB };
+  Writer();
+  ~Writer();
+
+  void Init(const ArrowSchema* schema) { Init(schema, ExportOptions()); }
+
+  void Init(const ArrowSchema* schema, const ExportOptions& options);
+
+  void Init(OutputType output_type, const ExportOptions& options);
+
+  void WriteGeography(const Geography& geog);
```

Comment:
> Do you also need a WriteNull()?

That's a good point (for spherely actually not yet, because we don't yet support missing values, but we should support it here in general). 
Add a separate method for it, or allow the `geog` to be a nullptr? (and check that inside `WriteGeography`?)